handlers/dry_run.go

```diff
@@ -0,0 +1,69 @@
+package handlers
+
+import (
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/jitsucom/eventnative/destinations"
+	"github.com/jitsucom/eventnative/enrichment"
+	"github.com/jitsucom/eventnative/events"
+	"github.com/jitsucom/eventnative/logging"
+	"github.com/jitsucom/eventnative/middleware"
+	"net/http"
+)
+
+const (
+	emptyDestinationIdMessage   = "destination_id must not be empty"
+	unknownDestinationIdMessage = "Destination with id=[%s] does not exist"
+)
+
+type DryRunHandler struct {
+	destinationService *destinations.Service
+	preprocessor       events.Preprocessor
+}
+
+func NewDryRunHandler(destinationService *destinations.Service, preprocessor events.Preprocessor) *DryRunHandler {
+	return &DryRunHandler{destinationService: destinationService, preprocessor: preprocessor}
+}
+
+func (drh *DryRunHandler) Handle(c *gin.Context) {
+	iface, ok := c.Get(middleware.TokenName)
+	if !ok {
+		logging.Error("Token wasn't found in context")
```

Comment:
It is the SystemError. If a token doesn't exist in the context - middleware wasn't called

Reply:
👍